core/services/relay/evm/read/filter.go

```diff
@@ -68,27 +100,35 @@ func (r *syncedFilter) SetFilter(filter logpoller.Filter) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
+	r.dirty = true
+
 	r.filter = filter
 }
 
 func (r *syncedFilter) SetName(name string) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
+	r.dirty = true
+
 	r.filter.Name = name
 }
 
 func (r *syncedFilter) AddAddress(address common.Address) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
+	r.dirty = true
+
 	r.filter.Addresses = append(r.filter.Addresses, address)
```

Comment:
Seems like we should have some sort of check here to make sure the address isn't already in the list (or perhaps, use a set instead of a list?).

I'm not sure how often we unregister and re-register the same filter, but if it's a lol then we'll keep appending more and more copies of the same address onto the list.

Reply:
Prior to this PR yes, but with the introduction of this PR it becomes possible. I'll comment above on the place where the check is removed.

Reply:
That said, it doesn't pose any serious problem so if you want to leave it as-is I'm okay with that.